### PR TITLE
Add style guide item on autolinks

### LINKED ISSUE #585: Item about autolinks in style guide
The style guide should have an item about autolink syntax. See #584 for discussion.

### PR DESCRIPTION
**Pull request by mattias-p**
## Purpose

Document how to keep our Markdown documentation compatible with tools other than Github.

## Context

Fixes #585.

## Changes

Adds an item to the style guide.

## How to test this PR

N/A

### CODE REVIEW

File: docs/internal-documentation/maintenance/StyleGuide.md [hunk 1]
```diff
@@ -14,3 +14,13 @@ specified below.
 * For variable names in test case specifications each word is capitalized (e.g.
   "Domain Part"). If a word is an abbreviation, it is usually written with all
   capitals, e.g. "DNS".
+* When linking to other documents and you want to include the URL in the
+  rendered text, make sure to put angle brackets around the URL, i.e. use the
+  [CommonMark autolink] syntax.
+
+  If you forget the angle brackets the URL it is still render as a link on Github
+  (per [GFM autolink] syntax), but other tools (e.g. link checking tools) are
+  likely to not recognize it as a link.
```
Comment (matsduf) [inline docs/internal-documentation/maintenance/StyleGuide.md:23]: This kind of argument can be left out. It should be what, not why.

**Review by pnax — DISMISSED** [review]
I like the purpose of this suggestion.

File: docs/internal-documentation/maintenance/StyleGuide.md [hunk 2]
```diff
@@ -14,3 +14,13 @@ specified below.
 * For variable names in test case specifications each word is capitalized (e.g.
   "Domain Part"). If a word is an abbreviation, it is usually written with all
   capitals, e.g. "DNS".
+* When linking to other documents and you want to include the URL in the
+  rendered text, make sure to put angle brackets around the URL, i.e. use the
+  [CommonMark autolink] syntax.
```
[the remark this answers is not in the document]
Comment (mattias-p) [reply docs/internal-documentation/maintenance/StyleGuide.md:19]: I added an example.

File: docs/internal-documentation/maintenance/StyleGuide.md (hunk 1 above)
Comment (mattias-p) [reply docs/internal-documentation/maintenance/StyleGuide.md:24]: When you apply the rule `what` is important, but when you change the rule `why` is important. I found the `why` to be non-obvious so I thought better to spell it out.

Comment (matsduf) [reply docs/internal-documentation/maintenance/StyleGuide.md:24]: OK, but could the "why" be shorter? E.g.
>If the angle brackets are not included, the URL it is render as a link on Github ([GFM autolink] syntax), but other tools may not recognize it as a link.

File: docs/internal-documentation/maintenance/StyleGuide.md [hunk 3]
```diff
@@ -14,3 +14,14 @@ specified below.
 * For variable names in test case specifications each word is capitalized (e.g.
   "Domain Part"). If a word is an abbreviation, it is usually written with all
   capitals, e.g. "DNS".
+* When linking to other documents and you want to include the URL in the
+  rendered text, make sure to put angle brackets around the URL, i.e. use the
```
Comment (matsduf) [inline docs/internal-documentation/maintenance/StyleGuide.md:18]: Make it a rule, not a suggestion, i.e. remove "make sure to".

Also, I suggest that the text does not speak to the reader, i.e. replace "you" with passive wording.

**Review by tgreenx — COMMENTED** [review]
LGTM, can approve when Mats comments are addressed.

**Review by pnax — APPROVED** [review]

**Pull request merged**